ASV44/ChatMessageBroker
# Architectural refactoring

### Background

Initial implementation of project have bad design and does not follow software development and language guidelines. Project have to be refactored for introducing clear separation of concerns between implemented logic and to implement architectural design.

### What has been done?
1. Remove unnecessary use of pointer in broker and client apps 
2. Add abstractions for decoupling network connection and messages decoding / encoding
3. Separate broker logic in dedicated components
4. Rewrite several methods to be create extensible modules and services
 
### How to test?
1. Build project by running `make` command
2. Run broker app and 2 instance of client app
3. Send messages between clients and execute all supported workspace commands
4. Make sure broker app manage all communication properly

### Checklist
